ui/src/App/index.jsx

```diff
@@ -135,80 +144,45 @@ export default class App extends Component {
       initSentry({ dsn: process.env.SENTRY_DSN });
     }
 
-    this.state = state;
-  }
-
-  static getDerivedStateFromError(error) {
-    return { error };
-  }
-
-  handleUserChanged = user => {
-    this.setState({
-      auth: {
-        ...this.state.auth,
-        user,
-      },
-    });
-  };
-
-  componentWillUnmount() {
-    this.authController.off('user-changed', this.handleUserChanged);
-  }
-
-  async componentDidMount() {
-    const themeType = await db.userPreferences.get('theme');
+    getThemeType();
 
-    this.authController.loadUser();
+    authController.loadUser();
 
-    if (themeType === 'light') {
-      this.setState({ theme: theme.lightTheme });
-    }
-  }
-
-  authorize = async user => {
-    this.authController.renew(user);
-  };
-
-  unauthorize = () => {
-    this.authController.setUser(null);
-  };
+    return () => {
+      authController.off('user-changed', handleUserChanged);
+    };
+  }, []);
 
-  toggleTheme = () => {
-    this.setState({
-      theme:
-        this.state.theme.palette.type === 'dark'
-          ? theme.lightTheme
-          : theme.darkTheme,
-    });
+  const toggleTheme = () => {
     const newTheme =
-      this.state.theme && this.state.theme.palette.type === 'dark'
+      themeState && themeState.palette.type === 'dark'
         ? theme.lightTheme
         : theme.darkTheme;
 
-    db.userPreferences.put(newTheme.palette.type, 'theme');
-    this.setState({ theme: newTheme });
+    setThemeState(newTheme);
   };
 
-  componentDidCatch(error, errorInfo) {
```

Comment:
Is there any possibility for us to use https://github.com/bvaughn/react-error-boundary?

Reply:
when i install this it does not show in package.json and getting this error:
 `Module not found: Error: Can't resolve 'react-error-boundary' in 'D:\OSS-issues\taskcluster_fork\taskcluster\ui\src\App'`

Shared a screenshot of the cli logs for installing packages in a comment above maybe its failing somehow.

Reply:
Since the issue with installing packages is resolved, this is working as expected i think. I have added this in a new commit.